### PR TITLE
Feature conector wfp

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

File: connector/sources/humanitarian_response.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        results = []
+        url = self.URL
+        if params.get('country'):
+            url = self.URL + '/loc/' + params['country']
+        resp = requests.get(url, params={})
+        soup = Soup(resp.text, 'html.parser')
+        contents = soup.find('div', {'id': 'content'}).find('tbody')
+        for row in contents.findAll('tr'):
+            tds = row.findAll('td')
+            print(tds[0])
```
Comment (bibekdahal) [inline connector/sources/humanitarian_response.py]: remove print()

File: connector/sources/wpf.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        results = []
+        if page:
+            params['page'] = page
+        resp = requests.get(self.URL, params=params)
+
+        soup = Soup(resp.text, 'html.parser')
+        contents = soup.find('div', {'class': 'view-content'})
+        if not contents:
+            return results, len(results)
+        # iterate and get loops
```
Comment (bibekdahal) [inline connector/sources/wpf.py]: Are you sure this is `loops` and not `leads`?

File: connector/sources/acaps_briefing_notes.py [hunk 3]
```diff
@@ -143,28 +145,24 @@ def fetch(self, params, offset=None, limit=None):
...
+                'div', {'class': 'field-item'}
+            ).find('a')
+            data = Lead(
+                # FIXME: use proper key
+                id=link['href'],
+                title=title.strip(),
+                published_on=date.date(),
+                url=link['href'],
+                source='Briefing Notes',
+                source_type='',  # FIXME source_type = website
```
Comment (bibekdahal) [inline connector/sources/acaps_briefing_notes.py]: Set `source_type = Lead.WEBSITE`

File: connector/sources/humanitarian_response.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            title = tds[0].find('a').get_text().strip()
+            datestr = tds[3].get_text().strip()
+            date = datetime.strptime(datestr, '%m/%d/%Y')
+            url = tds[4].find('a')['href']
+            data = Lead(
+                title=title,
+                published_on=date.date(),
+                url=url,
+                source='Humanitarian Response',
```
Comment (bibekdahal) [inline connector/sources/humanitarian_response.py:81]: Set `source_type = Lead.WEBSITE`

File: connector/sources/pdna.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            if name.strip() == country.strip():
+                # add as lead
+                url = elem['href']
+                if url[0] == '/':  # means relative path
+                    url = self.website + url
+                data = Lead(
+                    title=title.strip(),
+                    url=url,
+                    source='PDNA portal',
+                    source_type='',
```
Comment (bibekdahal) [inline connector/sources/pdna.py]: Set `source_type = Lead.WEBSITE`

File: connector/sources/research_center.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            tds = row.findAll('td')
+            title = tds[0].get_text().replace('_', ' ')
+            date = tds[1].find('span').attrs['content'][:10]  # just date str  # noqa
+            date = datetime.datetime.strptime(date, '%Y-%m-%d')
+            url = tds[0].find('a').attrs['href']
+            data = Lead(
+                title=title.strip(),
+                published_on=date.date(),
+                url=url,
+                source="Research Resource Center",
```
Comment (bibekdahal) [inline connector/sources/research_center.py:86]: Set `source_type = Lead.WEBSITE`

File: connector/sources/wpf.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        for row in contents.findAll('div', {'class': 'views-row'}):
+            content = row.find('h3').find('a')
+            title = content.get_text()
+            url = content['href']
+            data = Lead(
+                title=title.strip(),
+                url=url,
+                source='WFP Assessments',
```
Comment (bibekdahal) [inline connector/sources/wpf.py:178]: Set `source_type = Lead.WEBSITE`

**Review by bibekdahal — CHANGES_REQUESTED** [review]
If lead is of type website (that is have url), then set `source_type = Lead.WEBSITE`.

File: connector/sources/wpf.py (hunk 2 above)
Comment (bewakes) [reply connector/sources/wpf.py]: Lol.

**Comment by bewakes** [comment]
I've fixed them.

**Pull request merged**